libraries/Microsoft.Bot.Builder.Core.Extensions/FileTranscriptStore.cs

```diff
@@ -0,0 +1,216 @@
+        }
+
+        /// <summary>
+        /// Log an activity to the transcript
+        /// </summary>
+        /// <param name="activity">activity to log</param>
+        /// <returns></returns>
+        public async Task LogActivity(IActivity activity)
+        {
+            if (activity == null)
+                throw new ArgumentNullException("activity cannot be null for LogActivity()");
+
+            var activityPath = GetActivityPath(activity);
+
+            try
+            {
+                File.WriteAllText(activityPath, JsonConvert.SerializeObject(activity, jsonSettings));
+            }
+            catch (DirectoryNotFoundException)
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(activityPath));
+                File.WriteAllText(activityPath, JsonConvert.SerializeObject(activity, jsonSettings));
+            }
+            File.SetCreationTimeUtc(activityPath, activity.Timestamp.Value.UtcDateTime);
+        }
+
+        /// <summary>
+        /// Get activity records for conversationId
+        /// </summary>
+        /// <param name="channelId"></param>
+        /// <param name="conversationId"></param>
+        /// <param name="continuationToken"></param>
+        /// <param name="startDate"></param>
+        /// <returns></returns>
+        public async Task<PagedResult<IActivity>> GetConversationActivities(string channelId, string conversationId, string continuationToken = null, DateTime startDate = default(DateTime))
+        {
+            if (channelId == null)
+                throw new ArgumentNullException($"missing {nameof(channelId)}");
+
+            if (conversationId == null)
+                throw new ArgumentNullException($"missing {nameof(conversationId)}");
+
+            var pagedResult = new PagedResult<IActivity>();
+            var conversationPath = GetConversationFolder(channelId, conversationId);
+            DirectoryInfo dir = new DirectoryInfo(conversationPath);
+
+            if (dir.Exists)
+            {
+                FileInfo[] files;
+                if (continuationToken != null)
+                {
+                    files = dir.EnumerateFiles("*.json")
+                        .OrderBy(fi => fi.CreationTimeUtc)
+                        .Where(fi => fi.CreationTimeUtc >= startDate)
+                        .SkipWhile(fi => fi.Name != continuationToken)
+                        .Skip(1)
+                        .Take(20)
+                        .ToArray();
+                }
+                else
+                {
+                    files = dir.EnumerateFiles("*.json")
+                        .OrderBy(fi => fi.CreationTimeUtc)
+                        .Where(fi => fi.CreationTimeUtc >= startDate)
+                        .Take(20)
+                        .ToArray();
+                }
+                if (files.Length == 20)
+                    pagedResult.ContinuationToken = files.Last().Name;
```

Comment:
I don't think this is correct. Given that your sort order is CreationTimeUTC, that seems like the token that you need to be returning as the continuation token.

Reply:
... although on 2nd thought, the file name is the more unique of the two, so as long as the sort order is constant (which it is), this should be fine. Disregard. :)

Reply:
Or order by filename which seems like the real bug here.  Your filename has the tick count so you should order by filename,

---
In reply to: [181209682](https://github.com/Microsoft/botbuilder-dotnet/pull/410#discussion_r181209682) [](ancestors = 181209682)